ipa-core/src/protocol/ipa_prf/malicious_security/prover.rs

```diff
@@ -117,27 +118,92 @@ where
         <λ as Add>::Output: Add<U1>,
         <<λ as Add>::Output as Add<U1>>::Output: ArrayLength,
         <λ as Add<U1>>::Output: ArrayLength,
+        I: IntoIterator<Item = F>,
+        J: IntoIterator<Item = F>,
+        I::IntoIter: ExactSizeIterator,
+        J::IntoIter: ExactSizeIterator,
     {
-        assert_eq!(self.u.len(), λ::USIZE); // We should pad with zeroes eventually
+        let mut u = u.into_iter();
+        let mut v = v.into_iter();
+
+        assert_eq!(u.len(), λ::USIZE); // We should pad with zeroes eventually
+        assert_eq!(v.len(), λ::USIZE); // We should pad with zeroes eventually
 
         // We need a table of size `λ + 1` since we add a random point at x=0
         let denominator = CanonicalLagrangeDenominator::<F, Sum<λ, U1>>::new();
         let lagrange_table = LagrangeTable::<F, Sum<λ, U1>, λ>::from(denominator);
```

Comment:
you could use compute proof here as well to have less redundant code. You can even use the same lagrange table by only submitting lambda minus 1 real u, v value to leave one entry space for the masks. Then for the final proof you just need to slightly adapt he proof generator/ u, v by appending the masks when calling generate proof the last time

Reply:
That's a good point. I'll think about this. Maybe I can just remove this function entirely.

Reply:
Oh man... I tried, I really did, but the trait bounds are WILD! I just couldn't figure out how to make the compiler understand that it should work.